pkg/engine/apicall/apiCall.go

```diff
@@ -60,10 +60,20 @@ func (a *apiCall) FetchAndLoad(ctx context.Context) ([]byte, error) {
 func (a *apiCall) Fetch(ctx context.Context) ([]byte, error) {
 	call, err := variables.SubstituteAllInType(a.logger, a.jsonCtx, a.entry.APICall)
 	if err != nil {
+		if call == nil && a.entry.APICall.Default != nil {
```

Comment:
When and why would `call` be `nil` here?

I understand the use case of returning the default value when the API call returns a nil, but do we want to mask a substitution error?

Reply:
fair point! You are right, we should not mask a substitution error.